src/go/rpk/pkg/cli/topic/consume.go

```diff
@@ -183,19 +202,42 @@ func (c *consumer) consume() {
 			if done {
 				return
 			}
-
 			pend, hasEnd := c.getPartitionEnd(p.Topic, p.Partition)
 			if hasEnd && pend < 0 {
 				return // reached end, still draining client
 			}
 
+			serdeCache := make(map[int]*serde.Serde)
```

Comment:
This cache should come at the top of the for loop, not be created per fetch. Just to confirm -- we don't have a similar problem on the produce side, do we?

Reply:
I thought I had it up there, good catch.

No, in produce we don't cache the serde since it's the same for all the records you are producing (one id per flag -> one serde for key or value)